htdocs/compta/facture/card.php

```diff
@@ -1508,7 +1508,7 @@
 
 								$TTotalByTva = array();
 								foreach ($srcobject->lines as &$line) {
-									if (!empty($line->special_code)) {
+									if (!empty($line->special_code) && empty($line->info_bits)) {
```

Comment:
@SylvainLegrand 
Is this fix only for external modules ? (special_code is not empty)
Maybe, you can fix it by using the "insert_discount" method of "facture" class in the external module or by adding a line without a special code ?

Reply:
Yes this PR is for lines with special code AND info-bit !

Reply:
ok for me @rycks

Reply:
ok for me too ! sorry i have to make an effort to understand the problem :) and an other effort thanks to lvessiller to understand the use of info-bits